src/app/utils/api.class.ts

```diff
@@ -0,0 +1,33 @@
+import axios, { type AxiosInstance, type AxiosRequestConfig } from 'axios';
+
+const instance = axios.create({
+  baseURL: 'http://localhost:3000/api/v1',
```

Comment:
perhaps let's make this an environment variable which is substitutable at runtime. We can follow this article as a reference https://blog.innei.ren/nextjs-runtime-env-and-build-once-deploy-many?locale=en.